reporter/base_reporter.go

```diff
@@ -97,8 +98,11 @@ func (*baseReporter) ReportMetrics(_ uint32, _ []uint32, _ []int64) {}
 func (*baseReporter) SupportsReportTraceEvent() bool { return true }
 
 func (b *baseReporter) ReportTraceEvent(trace *libpf.Trace, meta *TraceEventMeta) {
-	traceEventsMap := b.traceEvents.WLock()
-	defer b.traceEvents.WUnlock(&traceEventsMap)
+	if meta.Origin != support.TraceOriginSampling && meta.Origin != support.TraceOriginOffCPU {
```

Comment:
We don't need to propagate a trace that we're going to reject all the way here, but should instead fast-fail as early as possible (e.g. in `tracer/events.go`)

Reply:
Added a check in `loadBpfTrace()` with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/196/commits/407d5714615f28fca4b1bfab21182235484597d3 to skip handling of traces of unknown origin as early as possible in the user space side.
Besides that, I'd like to keep this check in `baseReporter` as it makes sure that traces are assigned correctly to their respective kind.